internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2310,6 +2307,11 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 			existing.Model.Properties.NetworkProfile.NetworkDataplane = pointer.To(managedclusters.NetworkDataplane(ebpfDataPlane))
 		}
 
+		if key := "network_profile.0.network_policy"; d.HasChange(key) {
+			networkPolicy := d.Get(key).(string)
+			existing.Model.Properties.NetworkProfile.NetworkPolicy = pointer.To(managedclusters.NetworkPolicy(networkPolicy))
+		}
+
```

Comment:
This isn't necessary, since this is taken care of by the expand function for the `network_profile` block
```suggestion
```